stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/plugin/logic/channel/internal/ChannelLogic.kt

```diff
@@ -216,11 +216,12 @@ internal class ChannelLogic(
     }
 
     private suspend fun runChannelQuery(request: WatchChannelRequest): Result<Channel> {
-        val offlineChannel = runChannelQueryOffline(request)
+        val offlineChannel = if (!request.isFilteringNewerMessages()) runChannelQueryOffline(request) else null
```

Comment:
Hey @leandroBorgesFerreira,

Can we move this check right into `runChannelQueryOffline` method? :)
I can we see we do exact same check in `onQueryChannelRequest`.